marconi-core/src/Marconi/Core/Transformer/WithTracer.hs

```diff
@@ -463,3 +477,29 @@ instance
   => Resetable (t m) event (WithTrace m indexer)
   where
   reset = resetVia unwrap
+
+-- | A wrapper to hide the focus of a prism
+data SomePrism e = forall a. SomePrism (APrism' e a)
+
+-- | A helper for logging `IndexerError`s in an `IndexerEvent` trace
+logIndexerError
+  :: (MonadIO m)
+  => Trace m (IndexerEvent point)
+  -> IndexerError
+  -> m ()
+logIndexerError = logError IndexerFailed [SomePrism _StopIndexer]
+
+-- | A helper that allows us to add an exclusion list of error constructors
+logError
+  :: forall m e' e
+   . (MonadIO m)
+  => (e' -> e)
+  -> [SomePrism e']
+  -> Trace m e
+  -> e'
+  -> m ()
+logError fromErr exclusionList tr err =
+  ( when (any (\(SomePrism p) -> isn't p err) exclusionList) $
```

Comment:
Is it wrong? I read it as "when one error of the exclusion list isn't my error, log my error" (which will always be the case if we provide 2 different errors in the exclusion list).

```suggestion
  ( unless (any (\(SomePrism p) -> is p err) exclusionList) $
```

Reply:
Good catch!